File: frontendapp/lib/models/movie.dart
class Movie {
  final int id;
  final String name;
  final String description;
  final int duration;
  final String releaseDate;
  final String posterUrl;
  final String bannerUrl;
  final String ageRating;
  final double voteAverage;
  final String? director;

  Movie({
    required this.id,
    required this.name,
    required this.description,
    required this.duration,
    required this.releaseDate,
    required this.posterUrl,
    required this.bannerUrl,
    required this.ageRating,
    required this.voteAverage,
    this.director
  });

  factory Movie.fromJson(Map<String, dynamic> json) {
    return Movie(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      duration: json['duration'],
      releaseDate: json['releasedate'],
      posterUrl: json['posterurl'],
      bannerUrl: json['bannerurl'],
      ageRating: json['agerating'],
      voteAverage: (json['voteaverage'] as num).toDouble(),
      director: json['director'],
    );
  }
}

File: backend/bookingmovieticketapp/app/Models/Movie.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Movie extends Model
{
    use HasFactory;
    protected $table = 'movies';
    protected $primaryKey = 'id';
    public $incrementing = false;
    protected $keyType = 'int';
    public $timestamps = false;
    protected $fillable = [
        'id',
        'name',
        'description',
        'duration',
        'releasedate',
        'posterurl',
        'bannerurl',
        'agerating',
        'voteaverage',
        'director'
    ];

    public function showtimes()
    {
        return $this->hasMany(Showtime::class, 'movieid');
    }

    public function ratings()
    {
        return $this->hasMany(Rating::class, 'movieid');
    }

    public function casts()
    {
        return $this->hasMany(Cast::class, 'movieid');
    }
}
